From 67750e9efb3ba425d9a760d21c358a4514031ddf Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Sun, 2 Oct 2022 12:36:15 +0200
Subject: [PATCH 44/52] Revert "block, bfq: don't disable wbt if
 CONFIG_BFQ_GROUP_IOSCHED is disabled"

This reverts commit e38ca1a9197680fe897c8145dffa47ff7e6877c9.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/bfq-iosched.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/block/bfq-iosched.c b/block/bfq-iosched.c
index 442363680..e6d7e6b01 100644
--- a/block/bfq-iosched.c
+++ b/block/bfq-iosched.c
@@ -7039,7 +7039,6 @@ static void bfq_exit_queue(struct elevator_queue *e)
 
 #ifdef CONFIG_BFQ_GROUP_IOSCHED
 	blkcg_deactivate_policy(bfqd->queue, &blkcg_policy_bfq);
-	wbt_enable_default(bfqd->queue);
 #else
 	spin_lock_irq(&bfqd->lock);
 	bfq_put_async_queues(bfqd, bfqd->root_group);
@@ -7048,6 +7047,7 @@ static void bfq_exit_queue(struct elevator_queue *e)
 #endif
 
 	blk_stat_disable_accounting(bfqd->queue);
+	wbt_enable_default(bfqd->queue);
 
 	kfree(bfqd);
 }
@@ -7192,9 +7192,7 @@ static int bfq_init_queue(struct request_queue *q, struct elevator_type *e)
 	/* We dispatch from request queue wide instead of hw queue */
 	blk_queue_flag_set(QUEUE_FLAG_SQ_SCHED, q);
 
-#ifdef CONFIG_BFQ_GROUP_IOSCHED
 	wbt_disable_default(q);
-#endif
 	blk_stat_enable_accounting(q);
 
 	return 0;
-- 
2.38.0.rc1.6.g4fd6c5e444

